File: LowLevelDesign/SOLID_Assignment_1/C#/Rules/Violation.cs
namespace Rules
{
    public class Violation
    {
        private string _violationMessage;
        public Violation(string message)
        {
            this._violationMessage = message;
        }

        public static Violation CreateViolation(string message)
        {
            return new Violation(message);
        }
    }
}

File: LowLevelDesign/SOLID_Assignment_1/lib/Rules/Violation.java
package LowLevelDesign.SOLID_Assignment_1.lib.Rules;

public class Violation {

    private final String message;

    public Violation(String message) {
        this.message = message;
    }

    // returns Voilation object with message
    // why static => so that we don't need to create object of Violation class to
    // call this method
    public static Violation of(String message) {
        return new Violation(message);
    }

}
